File: math/Prime/isPrime.js
//This function returns true if num is prime, else returns false.
const isPrime = (num) => {
    if (num === 1) {
        return false;
    } else {
       for (let i = 2; i <= Math.sqrt(num); i++) {
           if (isPrime(i)) {
                if (num % i === 0) {
                    return false;
                }
           }
       }
       return true;
    }
}

console.log('5 is a prime number: ', isPrime(5));       //5 is prime
console.log('6 is a prime number: ', isPrime(6));       //6 is not prime

File: math/Prime/isPrime.java
import java.util.Scanner;
public class Prime {
	public static void main(String[]args) {
		int num,flag=0;
		Scanner sc = new Scanner(System.in);
		num=sc.nextInt();
		for(int i=2;i<=(num-1)/2;i++){
			if(num%i==0) {
				flag=1;
			}
		}
		if(flag==0) {
			System.out.println("Number is prime");
		}
		else {
				System.out.println("Number is not prime");
			}
		sc.close();
	}

}
